File: pupil/test/features/support/page_objects/sign_in_page.rb
class SignInPage < SitePrism::Page
  set_url '/sign-in'

  element :heading, '.heading-xlarge', text: "Multiplication tables check"
  section :phase_banner, PhaseBanner, '.js-content .phase-banner'

  element :welcome_message, '.lede', text: "To start your multiplication tables check, enter the school password and pupil PIN provided by your teacher, then click ‘Sign in’."

  element :logo, '.organisation-logo'
  element :first_letter, '#last-name'
  element :day, '#dob-day'
  element :month, '#dob-month'
  element :pin, '.pin-entry'
  element :school_pin, '#school-pin'
  element :pupil_pin, '#pupil-pin'
  element :sign_in_button, '.button-start'
  element :auto_complete_off, "form[autocomplete='off']"

  def login(pin_school, pin_pupil=nil)
    school_pin.set pin_school
    pupil_pin.set pin_pupil
  end

end

File: test/page_objects/admin/signInPage.js
// Page class for SignIn page
/* global element by */

class SignInPage {
  constructor () {
    this.heading = element(by.css('.heading-xlarge'))
    this.instructions = element(by.css('p.lede'))
    this.usernameField = element(by.css('#username'))
    this.passwordField = element(by.css('#password'))
    this.signIn = element(by.css('input[value="Sign in"]'))
  }

  enterCredentials (username, password) {
    this.usernameField.sendKeys(username)
    this.passwordField.sendKeys(password)
  }

  login (username, password) {
    expect(this.usernameField.isPresent()).to.eventually.be.true
    this.enterCredentials(username, password)
    this.signIn.click()
  }
}

module.exports = new SignInPage()
